src/app/containers/CpsFeaturesAnalysis/index.jsx

```diff
@@ -22,24 +22,33 @@ const PromoListComponent = ({ promoItems, dir }) => {
   const { serviceDatetimeLocale } = useContext(ServiceContext);
   const viewRef = useViewTracker(eventTrackingData.block);
   const { isAmp } = useContext(RequestContext);
+  const { enabled: frostedPromoEnabled, value: frostedPromoCount } =
+    useToggle('frostedPromo');
 
-  const StoryPromo = isAmp || isLive() ? _StoryPromo : FrostedGlassPromo;
+  const selectComponent = index => {
+    if (isAmp) return StoryPromo;
+    if (!frostedPromoEnabled) return StoryPromo;
+    return index + 1 <= frostedPromoCount ? FrostedGlassPromo : StoryPromo;
+  };
 
   return (
     <StoryPromoUl>
-      {promoItems.map((item, index) => (
-        <StoryPromoLi key={item.id || item.uri} ref={viewRef}>
-          <StoryPromo
-            item={item}
-            index={index}
-            dir={dir}
-            displayImage
-            displaySummary={false}
-            serviceDatetimeLocale={serviceDatetimeLocale}
-            eventTrackingData={eventTrackingData}
-          />
-        </StoryPromoLi>
-      ))}
+      {promoItems.map((item, index) => {
+        const Component = selectComponent(index);
```

Comment:
For readability maybe renaming this to `StoryPromoComponent` instead of the generic `Component`?

Reply:
```suggestion
        const PromoComponent = selectComponent(promoIndex);
```